lab_3_generate_by_ngrams/main.py

```diff
@@ -331,6 +537,21 @@ def continue_sequence(
 
         In case of corrupt input arguments or unexpected behaviour of methods used return None.
         """
+        if not (isinstance(sequence, tuple)
+                and isinstance(next_tokens, list)
+                and isinstance(sequence_candidates, dict) and sequence
+                and next_tokens and sequence_candidates and len(next_tokens) <= self._beam_width
+                and sequence in sequence_candidates
+        ):
+            return None
+
+        for token in next_tokens:
+            sequence_candidates[sequence + (token[0],)] = \
```

Comment:
Please beware of mutation, dictionary is mutable and this way you change the object outside of the method